server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -111,6 +114,19 @@
 
     public static final int MAX_INDEX_NAME_BYTES = 255;
 
+    /**
+     * These index patterns will be converted to hidden indices, at which point they should be removed from this list.
+     */
+    private static final CharacterRunAutomaton DOT_INDICES_EXCLUSIONS = new CharacterRunAutomaton(Regex.simpleMatchToAutomaton(
+        ".slm-history-*",
+        ".watch-history-*",
+        ".ml-anomalies-*",
+        ".ml-notifications-*",
+        ".ml-annotations*",
+        ".data-frame-notifications-*",
+        ".transform-notifications-*"
+    ));
```

Comment:
Not sure this is the best place for this, but it's easy to find and easy to move if we come up with a better location.